src/api/handlers.c

```diff
@@ -11,176 +11,22 @@
 // Contributors:
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
+
 #include "zenoh-pico/api/handlers.h"
 
-#include "zenoh-pico/api/macros.h"
-#include "zenoh-pico/protocol/core.h"
+#include "zenoh-pico/net/memory.h"
 #include "zenoh-pico/system/platform.h"
 
-// -- Channel
-z_owned_sample_channel_t z_owned_sample_channel_null() {
-    z_owned_sample_channel_t ch = {.send = NULL, .recv = NULL};
-    return ch;
-}
-
-int8_t z_closure_owned_sample_call(z_owned_closure_owned_sample_t *recv, z_owned_sample_t *dst) {
-    int8_t res = _Z_ERR_GENERIC;
-    if (recv != NULL) {
-        res = (recv->call)(dst, recv->context);
-    }
-    return res;
+// -- Sample
+void _z_owned_sample_move(z_owned_sample_t *dst, const z_owned_sample_t *src) {
+    memcpy(dst, src, sizeof(z_owned_sample_t));
 }
 
-// -- Ring
-void z_sample_channel_ring_push(const z_sample_t *src, void *context) {
-    if (src == NULL || context == NULL) {
-        return;
-    }
-
-    z_owned_sample_ring_t *r = (z_owned_sample_ring_t *)context;
+z_owned_sample_t *_z_sample_to_owned_ptr(const _z_sample_t *src) {
     z_owned_sample_t *dst = (z_owned_sample_t *)zp_malloc(sizeof(z_owned_sample_t));
-    if (dst == NULL) {
-        return;
-    }
-    *dst = z_sample_to_owned(src);
-
-#if Z_FEATURE_MULTI_THREAD == 1
-    zp_mutex_lock(&r->_mutex);
-#endif
-    _z_owned_sample_ring_push_force_drop(&r->_ring, dst);
-#if Z_FEATURE_MULTI_THREAD == 1
-    zp_mutex_unlock(&r->_mutex);
-#endif
-}
-
-int8_t z_sample_channel_ring_pull(z_owned_sample_t *dst, void *context) {
-    int8_t ret = _Z_RES_OK;
-
-    z_owned_sample_ring_t *r = (z_owned_sample_ring_t *)context;
-
-#if Z_FEATURE_MULTI_THREAD == 1
-    zp_mutex_lock(&r->_mutex);
-#endif
-    z_owned_sample_t *src = _z_owned_sample_ring_pull(&r->_ring);
-#if Z_FEATURE_MULTI_THREAD == 1
-    zp_mutex_unlock(&r->_mutex);
-#endif
-
-    if (src == NULL) {
-        *dst = z_sample_null();
-    } else {
-        memcpy(dst, src, sizeof(z_owned_sample_t));
-    }
-    return ret;
-}
-
-z_owned_sample_channel_t z_sample_channel_ring_new(size_t capacity) {
-    z_owned_sample_channel_t ch = z_owned_sample_channel_null();
-
-    z_owned_sample_ring_t *ring = (z_owned_sample_ring_t *)zp_malloc(sizeof(z_owned_sample_ring_t));
-    if (ring != NULL) {
-        int8_t res = _z_owned_sample_ring_init(&ring->_ring, capacity);
-#if Z_FEATURE_MULTI_THREAD == 1
-        res = zp_mutex_init(&ring->_mutex);
-#endif
-        if (res == _Z_RES_OK) {
-            z_owned_closure_sample_t send = z_closure(z_sample_channel_ring_push, NULL, ring);
-            ch.send = send;
-            z_owned_closure_owned_sample_t recv = z_closure(z_sample_channel_ring_pull, NULL, ring);
-            ch.recv = recv;
-        } else {
-            zp_free(ring);
-        }
+    if (dst && src) {
```

Comment:
In zenoh-pico we prefer explicit pointer checks, i.e. `if (dst != NULL && src != NULL)`